chips/msp432/src/cs.rs

```diff
@@ -0,0 +1,308 @@
+        /// HSMCLK ready status, indicates if the clock is stable after a change in the frequency/divider settings
+        HSMCLK_READY OFFSET(26) NUMBITS(1),
+        /// SMCLK ready status, indicates if the clock is stable after a change in the frequency/divider settings
+        SMCLK_READY OFFSET(27) NUMBITS(1),
+        /// BCLK ready status, indicates if the clock is stable after a change in the frequency/divider settings
+        BCLK_READY OFFSET(28) NUMBITS(1)
+    ],
+    /// Interrupt enable register
+    CSIE [
+        /// LFXT oscillator fault flag
+        LFXTIE OFFSET(0) NUMBITS(1),
+        /// HFXT oscillator fault flag
+        HFXTIE OFFSET(1) NUMBITS(1),
+        /// DCO external resistor open circuit fault flag
+        DCOR_OPNIE OFFSET(6) NUMBITS(1),
+        /// LFXT start fault counter
+        FCNTLFIE OFFSET(8) NUMBITS(1),
+        /// HFXT start fault counter
+        FCNTHFIE OFFSET(9) NUMBITS(1)
+    ],
+    /// Interrupt flag register
+    CSIFG [
+        /// LFXT oscillator fault flag
+        LFXTIFG OFFSET(0) NUMBITS(1),
+        /// HFXT oscillator fault flag
+        HFXTIFG OFFSET(1) NUMBITS(1),
+        /// DCO external resistor open circuit fault flag
+        DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        /// LFXT start fault counter
+        FCNTLFIFG OFFSET(8) NUMBITS(1),
+        /// HFXT start fault counter
+        FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    /// iIterrupt clear register
+    CSCLRIFG [
+        /// LFXT oscillator fault flag
+        LFXTIFG OFFSET(0) NUMBITS(1),
+        /// HFXT oscillator fault flag
+        HFXTIFG OFFSET(1) NUMBITS(1),
+        /// DCO external resistor open circuit fault flag
+        DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        /// LFXT start fault counter
+        FCNTLFIFG OFFSET(8) NUMBITS(1),
+        /// HFXT start fault counter
+        FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    /// Interrupt set/assert register
+    CSSETIFG [
+        /// LFXT oscillator fault flag
+        SET_LFXTIFG OFFSET(0) NUMBITS(1),
+        /// HFXT oscillator fault flag
+        SET_HFXTIFG OFFSET(1) NUMBITS(1),
+        /// DCO external resistor open circuit fault flag
+        SET_DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        /// LFXT start fault counter
+        SET_FCNTLFIFG OFFSET(8) NUMBITS(1),
+        /// HFXT start fault counter
+        SET_FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    /// DCO external resistor calibration 0 register
+    CSDCOERCAL0 [
+        /// DCO temperature compensation calibration
+        DCO_TCCAL OFFSET(0) NUMBITS(1),
+        /// DCO frequency calibration for DCO frequency range (DCORSEL) 0 to 4
+        DCO_FCAL_RSEL04 OFFSET(16) NUMBITS(10)
+    ],
+    /// DCO external resistor calibration 1 register
+    CSDCOERCAL1 [
+        /// DCO frequency calibration for DCO frequency range (DCORSEL) 5
+        DCO_FCAL_RSEL5 OFFSET(0) NUMBITS(10)
+    ]
+];
+
+pub struct ClockSystem {
+    registers: StaticRef<CsRegisters>,
```

Comment:
```suggestion
```

With the PeripheralManager interface, this structure should not hold a reference to the MMIO registers. Register access needs to go through the peripheral manager instead.

Reply:
Ahh ok, now I get it, thank you!
I think I use the `PeripheralMangager` now in the correct way